xchange-core/src/main/java/org/knowm/xchange/service/trade/TradeService.java

```diff
@@ -123,6 +124,27 @@ default String placeStopOrder(StopOrder stopOrder) throws IOException {
     throw new NotYetImplementedForExchangeException();
   }
 
+  /**
+   * Replaces an existing {@link LimitOrder} in a single atomic transaction.
+   * <br>This usually involves canceling the existing order and placing a new one, of the same
+   * {@link OrderType} but with an updated amount and/or price.
+   *
+   * @param replaceOrderId - the order ID to be replaced. When null is returned it means the atomic
+   * transaction has failed
+   * @param newLimitOrder - the new {@link LimitOrder}
+   * @return the new order ID
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   *     request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   *     requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   *     requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  default String replaceLimitOrder(String replaceOrderId, LimitOrder newLimitOrder) throws IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
```

Comment:
most exchanges do not implement this so it should not be in generic interfaces

https://github.com/knowm/XChange/wiki/Generic-feature-candidates

Reply:
I will remove the specific commit that adds replaceLimitOrder.

Reply:
I can remove the specific commit that adds replaceLimitOrder, but as its already done I think it's pretty much going backward because of "a bigger problem" you always point to.
@timmolter - can you please add your opinion?

Reply:
the problem is that you are adding a feature that is not generic to an generic exchange interface, non-generic features should go to non-generic *Raw classes  where you can access exchange specific features, there is no sense in having interfaces for methods that throw NotYetImplementedForExchangeException in most implementations

Reply:
ReplaceLimitOrder code is now removed. Resolving this conversation.